src/ucs/datastruct/callbackq.c

```diff
@@ -403,7 +403,7 @@ ucs_status_t ucs_callbackq_init(ucs_callbackq_t *cbq)
     ucs_callbackq_priv_t *priv = ucs_callbackq_priv(cbq);
     unsigned idx;
 
-    for (idx = 0; idx < UCS_CALLBACKQ_FAST_COUNT; ++idx) {
+    for (idx = 0; idx < UCS_CALLBACKQ_FAST_COUNT + 1; ++idx) {
```

Comment:
@evgeny-leksikov callbackq dispatch works under the assumption that the last element callback is NULL, because this is the condition to stop the `fast_elems` iteration. So the last element in the array has to be reserved and remain NULL. Decrementing the constant UCS_CALLBACKQ_FAST_COUNT, but doing +1 when defining the array, does that.
In addition, another element in `fast_elems` has to be reserved to the slow-path proxy, in case it is needed. This is the reason for
 `#define UCS_CALLBACKQ_FAST_MAX (UCS_CALLBACKQ_FAST_COUNT - 1)`